### PR TITLE
gitea: 1.13.7 -> 1.14.0

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

ChangeLog: https://github.com/go-gitea/gitea/releases/tag/v1.14.0

According to the changelog this also has some new security fixes, so we should discuss whether we want to backport it to 20.09.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 119127` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitea</li>
  </ul>
</details>

**Comment by stephank** [comment]
Result of `nixpkgs-review pr 119127` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitea</li>
  </ul>
</details>

**Review by aanderse — APPROVED** [review]
I read through the changelog and this LGTM :+1: 

I think we need to backport this, but continue to suggest that upstream should mention this isn't "stable" software and shouldn't be treated as such (ie. don't expect constant backports of security fixes, be prepared to continually update). We should likely mention this in the module system or something... The (not very good) alternative would be to suggest people stick to nixpkgs unstable for this package.

In conclusion I support backporting this ASAP.

**Comment by Ma27** [comment]
Available on stable as c3253cdab424b87ad8490d42f6585302b20bd39b. @aanderse do you have an idea where to place this? E.g. in the description of `services.gitea.enable`?

**Comment by aanderse** [comment]
Thanks @Ma27! I always appreciate your efforts on this :smile: 

I agree that `.enable` seems to be the most visible place. It might make sense to copy the text into the packages `longDescription` as well... but that might be a bit much.
